### ISSUE TITLE
Credo.Check.Refactor.Nesting False Negative: `for` comprehensions not counted

### ISSUE THREAD

**Issue by twinn** [rrrene/credo#1239]
## Description

The `Credo.Check.Refactor.Nesting` check does not count `for` comprehensions as a nesting level, while `Enum.map` and other constructs are counted. This allows deeply nested code using `for` to pass the check when equivalent code using `Enum.map` would fail.

## Example

The following code with 4 nested `for` comprehensions passes credo with no nesting warnings:

```elixir
defmodule ForNesting do
  def deeply_nested_for(list) do
    for a <- list do                    # Level 1
      for b <- a do                     # Level 2
        for c <- b do                   # Level 3
          for d <- c do                 # Level 4
            d * 2
          end
        end
      end
    end
  end
end
```

However, equivalent code using Enum.map correctly triggers the nesting warning:

```elixir
defmodule MapNesting do
  def deeply_nested_map(list) do
    Enum.map(list, fn a ->              # Level 1
      Enum.map(a, fn b ->               # Level 2
        Enum.map(b, fn c ->             # Level 3
          Enum.map(c, fn d ->           # Level 4
            d * 2
          end)
        end)
      end)
    end)
  end
end
```

```bash
[F] → Function body is nested too deep (max depth is 2, was 4).
```

### Expected Behavior

Both examples should trigger the nesting depth warning since they represent the same logical nesting structure.

### Environment

- Credo version: 1.7.15
- Elixir version: 1.19.1
- Erlang/OTP: 28

### Impact

This inconsistency allows developers to bypass nesting checks by using for comprehensions instead of Enum.map, which defeats the purpose of the readability/complexity check since both constructs have equivalent nesting complexity.

**Comment by rrrene** [comment]
Thx for reporting! Definitely a bug. :+1:

**Comment by rrrene** [comment]
@twinn It should be fixed on `master` 😀  

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by twinn** [comment]
@rrrene yep issue is fixed (and now I have a lot of issues to fix hahaha). Thank you so much!

**Comment by rrrene** [comment]
@twinn  This is now part of Credo [`1.7.16`](https://github.com/rrrene/credo/releases/tag/v1.7.16) 🎉
